src/discordtipbot/discordbot.cr

```diff
@@ -17,6 +17,7 @@ class DiscordBot
     @bot.cache = @cache
     @tip = TipBot.new(@config, @log)
     @active_users_cache = ActivityCache.new(10.minutes)
+    @presence_cache = PresenceCache.new
```

Comment:
I'm not sure this warrants its own type. You're not really doing anything special with the storage, `PresenceCache` is just a light wrapper around `Set(UInt64)`. The caching code could easily just be inside those `guild_create` and `presence_update` handlers. Unless you plan on doing more with `PresenceCache` later, might be best to just keep it simple.

To contrast with the other cache we wrote, that was necessary because we needed special time-based storage.

Reply:
ATM I'm not planning anything special, but I think it's a lot cleaner having it in it's own class (also makes the code look nicer 😄 (I probably should just be adding more comments etc))